qiskit/circuit/library/n_local/swaprz.py

```diff
@@ -0,0 +1,155 @@
+                See the Examples section of :class:`~qiskit.circuit.library.TwoLocal` for more
+                detail.
+            initial_state: An `InitialState` object to prepend to the circuit.
+            skip_unentangled_qubits: If True, the single qubit gates are only applied to qubits
+                that are entangled with another qubit. If False, the single qubit gates are applied
+                to each qubit in the Ansatz. Defaults to False.
+            skip_unentangled_qubits: If True, the single qubit gates are only applied to qubits
+                that are entangled with another qubit. If False, the single qubit gates are applied
+                to each qubit in the Ansatz. Defaults to False.
+            skip_final_rotation_layer: If True, a rotation layer is added at the end of the
+                ansatz. If False, no rotation layer is added. Defaults to True.
+            parameter_prefix: The parameterized gates require a parameter to be defined, for which
+                we use :class:`~qiskit.circuit.ParameterVector`.
+            insert_barriers: If True, barriers are inserted in between each layer. If False,
+                no barriers are inserted.
+            depth: Deprecated, use `reps` instead.
+            entangler_map: Deprecated, use `entanglement` instead. This argument now also supports
+                entangler maps.
+
+        Examples:
+            >>> swaprz = SwapRZ(3, reps=1, insert_barriers=True, entanglement='linear')
+            >>> print(swaprz)  # show the circuit
+            ┌──────────┐ ░ ┌────────────┐┌────────────┐                             ░ ┌──────────┐
+            ┤ Rz(θ[0]) ├─░─┤0           ├┤0           ├─────────────────────────────░─┤ Rz(θ[5]) ├
+            ├──────────┤ ░ │  Rxx(θ[3]) ││  Ryy(θ[3]) │┌────────────┐┌────────────┐ ░ ├──────────┤
+            ┤ Rz(θ[1]) ├─░─┤1           ├┤1           ├┤0           ├┤0           ├─░─┤ Rz(θ[6]) ├
+            ├──────────┤ ░ └────────────┘└────────────┘│  Rxx(θ[4]) ││  Ryy(θ[4]) │ ░ ├──────────┤
+            ┤ Rz(θ[2]) ├─░─────────────────────────────┤1           ├┤1           ├─░─┤ Rz(θ[7]) ├
+            └──────────┘ ░                             └────────────┘└────────────┘ ░ └──────────┘
+
+            >>> swaprz = SwapRZ(2, reps=1)
+            >>> qc = QuantumCircuit(2)  # create a circuit and append the RY variational form
+            >>> qc.cry(0.2, 0, 1)  # do some previous operation
+            >>> qc.compose(swaprz, inplace=True)  # add the swaprz
+            >>> qc.draw()
+                            ┌─────────────────┐┌──────────────────┐┌─────────────────┐
+            q_0: ─────■─────┤ Circuit56(θ[0]) ├┤0                 ├┤ Circuit61(θ[3]) ├
+                 ┌────┴────┐├─────────────────┤│  Circuit59(θ[2]) │├─────────────────┤
+            q_1: ┤ Ry(0.2) ├┤ Circuit57(θ[1]) ├┤1                 ├┤ Circuit62(θ[4]) ├
+                 └─────────┘└─────────────────┘└──────────────────┘└─────────────────┘
```

Comment:
This was due to `append(....to_instruction())`. I changed the code to use `compose` and to hold circuits as building blocks and not instructions and now the circuit names do not get lost. However also the `Rxx+yy` got lost because the `RXX` and `RYY` operations get appended immediately.